src/tools/lib/python/picongpu/plugins/energy_histogram.py

```diff
@@ -0,0 +1,160 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2017 PIConGPU contributors
+Authors: Axel Huebl
+License: GPLv3+
+"""
+
+import numpy as np
+import os
+
+
+class EnergyHistogram(object):
+    """
+    Data Reader for the Energy Histogram Plugin.
+    """
+
+    def __init__(self, run_directory):
+        """
+        Parameters
+        ----------
+        simulation_directory : string
+            path to the run directory of PIConGPU
+            (the path before ``simOutput/``)
+        """
+        if run_directory is None:
+            raise ValueError('The run_directory parameter can not be None!')
+
+        self.run_directory = run_directory
+        self.data_file_suffix = "_energyHistogram.dat"
+
+    def get_data_path(self, species):
+        """
+        Return the path to the underlying data file.
+
+        Parameters
+        ----------
+        species : string
+            short name of the particle species, e.g. 'e' for electrons
+            (defined in ``speciesDefinition.param``)
+
+        Returns
+        -------
+        A string with a path.
+        """
+        if species is None:
+            raise ValueError('The species parameter can not be None!')
+
+        sim_ouput_dir = os.path.join(self.run_directory, "simOutput")
+        if not os.path.isdir(sim_ouput_dir):
+            raise IOError('The simOutput/ directory does not exist inside '
```

Comment:
OPTIONAL:
This error message could be clearer (just stating `simOutput` was not found) if the existence of the base directory was checked before.